setup.go

```diff
@@ -491,6 +341,178 @@ func setupDatastore(cfg Config) (datastore.Batching, error) {
+		dht.BootstrapPeers(dht.GetDefaultBootstrapPeerAddrInfos()...),
+		dht.Mode(dht.ModeClient),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if cfg.DHTRouting == DHTStandard {
+		return standardClient, nil
+	}
+
+	if cfg.DHTRouting == DHTAccelerated {
+		fullRTClient, err := fullrt.NewFullRT(dhtHost, dht.DefaultPrefix,
+			fullrt.DHTOption(
+				dht.Validator(record.NamespacedValidator{
+					"pk":   record.PublicKeyValidator{},
+					"ipns": ipns.Validator{KeyBook: h.Peerstore()},
+				}),
+				dht.Datastore(ds),
+				dht.BootstrapPeers(dht.GetDefaultBootstrapPeerAddrInfos()...),
+				dht.BucketSize(20),
+			))
+		if err != nil {
+			return nil, err
+		}
+		return &bundledDHT{
+			standard: standardClient,
+			fullRT:   fullRTClient,
+		}, nil
+	}
+
+	return nil, fmt.Errorf("unknown DHTRouting option: %q", cfg.DHTRouting)
+}
+
+func setupCompositeRouting(delegatedRouters []routing.Routing, dht routing.Routing) routing.Routing {
+	// Default router is no routing at all: can be especially useful during tests.
+	var router routing.Routing
+	router = &routinghelpers.Null{}
+
+	if len(delegatedRouters) == 0 && dht != nil {
+		router = dht
+	} else {
+		var routers []*routinghelpers.ParallelRouter
+
+		if dht != nil {
+			routers = append(routers, &routinghelpers.ParallelRouter{
+				Router:                  dht,
+				ExecuteAfter:            0,
+				DoNotWaitForSearchValue: true,
+				IgnoreError:             false,
+			})
+		}
+
+		for _, routingV1Router := range delegatedRouters {
+			routers = append(routers, &routinghelpers.ParallelRouter{
+				Timeout:                 15 * time.Second,
+				Router:                  routingV1Router,
+				ExecuteAfter:            0,
+				DoNotWaitForSearchValue: true,
+				IgnoreError:             true,
+			})
+		}
+
+		if len(routers) > 0 {
+			router = routinghelpers.NewComposableParallel(routers)
+		}
+	}
+
+	return router
+}
+
+func setupRouting(ctx context.Context, cfg Config, h host.Host, ds datastore.Batching, dhtRcMgr network.ResourceManager, bwc metrics.Reporter, dnsCache *cachedDNS) (routing.ContentRouting, routing.PeerRouting, routing.ValueStore, error) {
+	delegatedRouters, err := setupDelegatedRouting(cfg, dnsCache)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	dhtRouter, err := setupDHTRouting(ctx, cfg, h, ds, dhtRcMgr, bwc)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	router := setupCompositeRouting(delegatedRouters, dhtRouter)
+
+	var (
+		cr routing.ContentRouting = router
+		pr routing.PeerRouting    = router
+		vs routing.ValueStore     = router
+	)
+
+	// If we're using seed peering, we need to run a lighter Amino DHT for the
+	// peering routing. We need to run a separate DHT with the main host if
+	//  the shared host is disabled, or if we're not running any DHT at all.
+	if cfg.SeedPeering && (!cfg.DHTSharedHost || dhtRouter == nil) {
+		pr, err = dht.New(ctx, h,
+			dht.Datastore(ds),
+			dht.BootstrapPeers(dht.GetDefaultBootstrapPeerAddrInfos()...),
```

Comment:
:thought_balloon: we may want to make bootstrappers configurable for private deployments at some point, but ok with default for  now